android/src/main/java/com/polidea/flutter_ble_lib/delegate/DeviceConnectionDelegate.java

```diff
@@ -115,17 +131,28 @@ public void onError(final BleError error) {
                 });
     }
 
-    private void observeConnectionState(String deviceId, boolean emitCurrentValue, @NonNull final MethodChannel.Result result) {
+    private void observeConnectionState(final String deviceId, boolean emitCurrentValue, @NonNull final MethodChannel.Result result) {
         //emit current value if needed; rest is published automatically through connectToDevice()
 
         final SafeMainThreadResolver safeMainThreadResolver = new SafeMainThreadResolver<>(
                 new OnSuccessCallback<Boolean>() {
                     @Override
                     public void onSuccess(Boolean data) {
+                        String newData;
```

Comment:
Can we improve naming here? Data says nothing to me (maybe rename it to isConnected). Moreover there is no need to declare "String newData" at the top of the onSuccess callback. We can move it right above the place it is used - try block. We can also rename newData to connectionStateJson or sth like that. What about moving this parsing logic into this particular stream handler?